src/vm/tests/conversions.rs

```diff
@@ -230,3 +232,137 @@ fn test_simple_buff_to_uint_be() {
         .into()
     );
 }
+
+#[test]
+fn test_simple_string_to_int() {
+    let good1_test = r#"(string-to-int "-1")"#;
+    assert_eq!(Value::Int(-1), execute_v2(good1_test).unwrap().unwrap());
+
+    let good1_test = r#"(string-to-int u"-1")"#;
+    assert_eq!(Value::Int(-1), execute_v2(good1_test).unwrap().unwrap());
+
+    let no_args_test = r#"(string-to-int)"#;
+    assert_eq!(
+        execute_v2(no_args_test).unwrap_err(),
+        CheckErrors::IncorrectArgumentCount(1, 0).into()
+    );
+
+    let bad_value_error_test = r#"(string-to-int "a")"#;
+    assert_eq!(
+        execute_v2(bad_value_error_test).unwrap_err(),
+        CheckErrors::InvalidCharactersDetected.into()
+    );
+
+    let wrong_type_error_test = r#"(string-to-int 1)"#;
+    assert_eq!(
+        execute_v2(wrong_type_error_test).unwrap_err(),
+        CheckErrors::UnionTypeValueError(
+            vec![
+                TypeSignature::max_string_ascii(),
+                TypeSignature::max_string_utf8(),
+            ],
+            Value::Int(1)
+        )
+        .into()
+    );
```

Comment:
Can you add test coverage for _all_ ASCII characters, including the unprintable ones?  Also, can you try converting unprintable UTF-8 strings, or nonsensical UTF-8 strings like emojis and the like, into integers, and verify that they fail with `(err ...)` or `none`?

Reply:
I added some weird character examples, as well as an integer that has a very long length.